project-api/project-api-test/src/test/java/io/ballerina/projects/test/PackageResolutionTests.java

```diff
@@ -153,14 +153,70 @@ public void testProjectWithInvalidBuildFile() throws IOException {
 
         PackageCompilation compilation = loadProject.currentPackage().getCompilation();
 
+        boolean check = false;
         DependencyGraph<ResolvedPackageDependency> dependencyGraph = compilation.getResolution().dependencyGraph();
         for (ResolvedPackageDependency graphNode : dependencyGraph.getNodes()) {
             Collection<ResolvedPackageDependency> directDeps = dependencyGraph.getDirectDependencies(graphNode);
+            PackageManifest manifest = graphNode.packageInstance().manifest();
+            if (manifest.name().value().equals("package_o") && (manifest.version().toString().equals("1.0.2"))) {
+                check = true;
+            }
+        }
+
+        Assert.assertTrue(check);
+    }
+
+    @Test(description = "tests resoultion with toml dependency")
+    public void testBase() {
+
+        // Setup
+        BCompileUtil.compileAndCacheBala("projects_for_resolution_tests/package_o_1_0_0");
+        BCompileUtil.compileAndCacheBala("projects_for_resolution_tests/package_o_1_0_2");
+        BCompileUtil.compileAndCacheBala("projects_for_resolution_tests/package_o_1_1_0");
+
+        // Stage 1 : Package P without dep in Ballerina toml
+        Path projectDirPath = RESOURCE_DIRECTORY.resolve("package_p_withoutDep");
+
+        BuildOptions.BuildOptionsBuilder buildOptionsBuilder = BuildOptions.builder().setExperimental(true);
+        buildOptionsBuilder.setSticky(false);
+        BuildOptions buildOptions = buildOptionsBuilder.build();
+
+        Project loadProject = TestUtils.loadBuildProject(projectDirPath, buildOptions);
+        PackageCompilation compilation = loadProject.currentPackage().getCompilation();
+
+        boolean check = false;
+        DependencyGraph<ResolvedPackageDependency> dependencyGraph = compilation.getResolution().dependencyGraph();
+        for (ResolvedPackageDependency graphNode : dependencyGraph.getNodes()) {
```

Comment:
We can improve this to get rid of the boolean used just to verify the dependency exists in the graph.

```java
esolvedPackageDependency package_o = dependencyGraph.getNodes().stream().filter(node -> 
        node.packageInstance().manifest().name().toString().equals("package_o"))
.findFirst().orElseThrow()
```